waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/ASTQueryMapping.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (C) 2016-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.waggledance.mapping.model;
+
+import static org.apache.hadoop.hive.ql.parse.BaseSemanticAnalyzer.unescapeIdentifier;
+
+import static com.hotels.bdp.waggledance.parse.ASTNodeUtils.getChildren;
+import static com.hotels.bdp.waggledance.parse.ASTNodeUtils.getRoot;
+import static com.hotels.bdp.waggledance.parse.ASTNodeUtils.replaceNode;
+
+import java.util.Stack;
+
+import org.antlr.runtime.CommonToken;
+import org.antlr.runtime.Token;
+import org.apache.hadoop.hive.ql.parse.ASTNode;
+import org.apache.hadoop.hive.ql.parse.HiveParser;
+import org.apache.hadoop.hive.ql.parse.ParseException;
+import org.apache.hadoop.hive.ql.parse.ParseUtils;
+
+import com.hotels.bdp.waggledance.api.WaggleDanceException;
+import com.hotels.bdp.waggledance.parse.ASTConverter;
+
+public enum ASTQueryMapping implements QueryMapping {
+
+  INSTANCE;
+
+  @Override
+  public String transformOutboundDatabaseName(MetaStoreMapping metaStoreMapping, String query) {
+    ASTNode root;
+    try {
+      root = ParseUtils.parse(query);
+    } catch (ParseException e) {
+      throw new WaggleDanceException(e.getMessage());
```

Comment:
Is there a reason for not adding the exception here? Generally I like being able to see the stack trace of the cause unless it doesn't add anything (which is rarely the case)

Reply:
I agree, I'll throw in the exception